1-tolean.py

```diff
@@ -493,12 +494,26 @@ def check_opt(opt, timeout, bitwidth, hide_progress):
 
   return True # succeeded, did not time out
 
+def sanitize_name(name):
+  renamed = name.replace(':', '_').replace('-','_').replace(' ','_')
+  #TODO: this should be a regex but I don't know the python syntax for it ATM
+  for s in ["(",")", "~", "&", ">"]:
+    pos = renamed.find(s)
+    if pos != -1:
+      renamed = renamed[:pos]
+  return renamed
 
 def print_as_lean(opt):
   name, pre, src, tgt, ident_src, ident_tgt, used_src, used_tgt, skip_tgt = opt
-  (src_str, src_state) = to_lean_prog(src, num_indent=2, skip=[])
-  (tgt_str, tgt_state) = to_lean_prog(tgt, num_indent=2, skip=[])
-
+  print("dbg> printing " + name + " as lean")
+  (src_str, src_state, src_bw) = to_lean_prog(src, num_indent=2, skip=[])
+  (tgt_str, tgt_state, tgt_bw) = to_lean_prog(tgt, num_indent=2, skip=[])
+  bitwidth = unify_bitwidths([src_bw, tgt_bw])
+  print("dbg> lhs bw: " + str(src_bw) + " rhs bw: " + str(tgt_bw) + " unified to: " + str(bitwidth))
+  #TODO: this is a bit of a drastic unification here... can probably be more principled
+  #But at least it is sound... and in fairness, Alive is somewhat imprecise here
+  src_str = src_str.replace(" w ", " " + str(bitwidth) + " ")
+  tgt_str = tgt_str.replace(" w ", " " + str(bitwidth) + " ")
```

Comment:
I feel obliged to note that I consider manipulating the stringified version of the AST to replace the bitwidth to be a total hack =)